### PR TITLE
Payment step cleanup

### PR DESCRIPTION
**Pull request by elia**
## Summary

Removes the two tier radio buttons to select between wallet payment sources and filling in from scratch.

It's predicated under the following assumptions:

- The previous system was confusing, with its interacting radio buttons which were presented on both sides without much consistency.
- The list gives precedence to wallet payments as they require less clicks and were already the default choice if present.
- If no wallet payments are available the first payment method is selected, according to the order set in admin.
- Since to the customer they're all payment methods they're treated as a single radio list.
- Both types of payments are now visually consistent, better communicating all the options to the customer.
- The payment details are wrapped in a fieldset that gets disabled whenever a payment method is unselected, this also works as a hook for the payment provider specific implementation (e.g. solidus_stripe) to react properly when selected/unselected.

This layout and decisions are also informed by multiple real world stores that got their payment step redesigned.

- depends on #301 

### Before
<details>
<summary>
Show original behavior
</summary>

https://user-images.githubusercontent.com/1051/219420532-c91ffe24-ea19-488b-84f4-8ba45145195e.mov

image
</details>

### After

image

https://user-images.githubusercontent.com/1051/219440642-112350bc-1893-459c-b4fd-c8f24aba4e44.mov

## Checklist

Check out our [PR guidelines](https://github.com/solidusio/.github/blob/master/CONTRIBUTING.md#pull-request-guidelines) for more details.

The following are mandatory for all PRs:

- [x] I have written a thorough PR description.
- [x] I have kept my commits small and atomic.
- [x] [I have used clear, explanatory commit messages](https://github.com/solidusio/.github/blob/main/CONTRIBUTING.md#writing-good-commit-messages).

The following are not always needed:

- 📖 I have updated the README to account for my changes.
- 📑 I have documented new code [with YARD](https://www.rubydoc.info/gems/yard/file/docs/Tags.md).
- 🛣️ I have opened a PR to update the [guides](https://github.com/solidusio/edgeguides).
- ✅ I have added automated tests to cover my changes.
- 📸 I have attached screenshots to demo visual changes.

### CODE REVIEW

**Comment by kennyadsl** [comment]
@elia still not reviewing but... a "before" screenshot would be helpful to quickly navigate what's changing, especially when the description of the PR is empty. 😬

**Comment by elia** [comment]
@kennyadsl yes 💯 agree, will probably do a short video demonstrating how broken it is

File: templates/app/views/checkouts/payment/_check.html.erb [hunk 1]
```diff
@@ -0,0 +1 @@
+<%= payment_method.class.model_name.human %>
```
Comment (kennyadsl) [inline templates/app/views/checkouts/payment/_check.html.erb:1]: I think Payment Method has a description field, we could use that.

File: templates/app/javascript/controllers/checkout_payment_controller.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import { Controller } from "@hotwired/stimulus"
```
Comment (kennyadsl) [inline templates/app/javascript/controllers/checkout_payment_controller.js:1]: ❤️

**Review by kennyadsl — APPROVED** [review]
Left one quick improvement idea, but overall: YES!

**Pull request merged**